streams/readable-streams/tee.any.js

```diff
@@ -371,6 +371,42 @@ promise_test(async t => {
 
 }, 'ReadableStream teeing: canceling branch1 should finish when original stream errors');
 
+promise_test(async () => {
+
+  const rs = new ReadableStream({});
+
+  const [branch1, branch2] = rs.tee();
+
+  const cancel1 = branch1.cancel();
+  await flushAsyncEvents();
+  const cancel2 = branch2.cancel();
+
+  await Promise.all([cancel1, cancel2]);
+
+}, 'ReadableStream teeing: canceling both branches in sequence with delay');
+
+promise_test(async t => {
+
+  const theError = { name: 'boo!' };
+  const rs = new ReadableStream({
+    cancel() {
+      throw theError;
+    }
+  });
+
+  const [branch1, branch2] = rs.tee();
+
+  const cancel1 = branch1.cancel();
+  await flushAsyncEvents();
+  const cancel2 = branch2.cancel();
+
+  await Promise.all([
+    promise_rejects_exactly(t, theError, cancel1),
+    promise_rejects_exactly(t, theError, cancel2)
+  ]);
+
+}, 'ReadableStream teeing: failing to cancel when canceling both branches in sequence with delay');
```

Comment:
I think some browsers might trip up on this test. The order of operations inside `ReadableStreamTee` is:
1. _cancel1Algorithm_ sets `canceled1 = true`
2. _cancel2Algorithm_ sets `canceled2 = true`
3. _cancel2Algorithm_ calls `ReadableStreamCancel`
4. `ReadableStreamCancel` calls `ReadableStreamClose`, which calls the _closeSteps_
5. `ReadableStreamCancel` calls `[[CancelSteps]]`, which calls `underlyingSource.cancel()`
6. _cancel2Algorithm_ resolves _cancelPromise_ with the promise from `ReadableStreamCancel` (which comes from `underlyingSource.cancel()`)

Previously, we would resolve _cancelPromise_ in step 4 (_closeSteps_), so we'd never see any rejections from `source.cancel()`. With whatwg/streams#1118, we will now resolve _cancelPromise_ in step 6 instead, so we *will* see those rejections.

Let's wait and see what the browser test results say. We might have to open some more browser bugs. 😁

Reply:
Okay, so [Chrome](https://wpt.fyi/results/streams/readable-streams/tee.any.html?diff&filter=ADC&run_id=5106808228151296&run_id=5735591440285696) already passes both of the new tests. [Firefox](https://wpt.fyi/results/streams/readable-streams/tee.any.html?diff&filter=ADC&run_id=4819103938248704&run_id=5650920253685760) and [Safari](https://wpt.fyi/results/streams/readable-streams/tee.any.html?diff&filter=ADC&run_id=5637437613146112&run_id=5669758181572608) pass the first test, but fail the second one.